engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/ForEachTransformer.scala

```diff
@@ -27,11 +27,13 @@ object ForEachTransformer extends CustomStreamTransformer with Serializable {
           .flatMap(ctx.lazyParameterHelper.lazyMapFunction(elements))
           .flatMap(
             (valueWithContext: ValueWithContext[util.Collection[AnyRef]], c: Collector[ValueWithContext[AnyRef]]) => {
```

Comment:
I don't know the topic fully, but `valueWithContext: ValueWithContext[util.Collection[AnyRef]]` declaration makes me think that there should be some Collection there. 

I think there's 2 types of situations where a null can end up here:
1. Someone enters a self-contained expression that evaluates to null
2. Someone enters a reference to a list coming from external system - like a list field from a kafka source - and that evaluates to null

For the 1. situation I think we should tackle this at parameter level. I don't see why it would be useful to have a foreach that doesn't do anything. So I would propose adding a NotNull validation to it.

For the 2. situation I think this we should signal this as exception. I think that if configurator expects a null in list field he should do a null check before? And if he omitted that I feel like administrator would like to know about that and communicate it to the configurator. Although I don't have practical experience in these roles so I'm just guessing. WDYT?

Also, do we have another way of handling nulls in our code at component implementation level?

Reply:
Remember that: `util.Collection[AnyRef]` is java, so null is posible here :)